frontend/src/App.tsx

```diff
@@ -3,36 +3,52 @@ import { render } from 'react-dom';
 import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
 import LinkItem from './components/LinkItem';
 import TokenPage from './FullPageRoutes/TokenPage';
+import RecordPage from './FullPageRoutes/RecordPage';
+import VideoListPage from './FullPageRoutes/VideoListPage';
+import { AppContext } from './utils/context';
 import ProfilePage from './FullPageRoutes/Profile';
 
 export type User = {
     username: string;
 } & any;
 
+//@TODO: Move links to common file
 function MainPage() {
     return (
         <ul>
             <LinkItem to='/Tokens' text='Tokens' />
+            <LinkItem to='/Record' text='Record' />
+            <LinkItem to='/Watch' text='Watch' />
             <LinkItem to='/Profile' text='Profile' />
         </ul>
     );
 }
 
 export default function App() {
     return (
-        <Router>
-            <Switch>
-                <Route path='/Tokens'>
-                    <TokenPage />
-                </Route>
-                <Route path='/Profile'>
-                    <ProfilePage />
-                </Route>
-                <Route path='/'>
-                    <MainPage />
-                </Route>
-            </Switch>
-        </Router>
+        <AppContext.Provider
+            value={{ focusedUser: React.useState<number>(null) }}
+        >
+            <Router>
+                <Switch>
+                    <Route path='/Tokens'>
+                        <TokenPage />
+                    </Route>
+                    <Route path='/Record'>
+                        <RecordPage />
+                    </Route>
+                    <Route path='/Watch'>
+                        <VideoListPage />
+                    </Route>
+                    <Route path='/Profile'>
+                        <ProfilePage />
+                    </Route>
+                    <Route path='/'>
+                        <MainPage />
+                    </Route>
+                </Switch>
+            </Router>
+        </AppContext.Provider>

```

Comment:
By wrapping the body `App.tsx` in `AppContext.Provider`, I've initialized the global state and applied it to the app context.